packages/dataviews/src/utils/normalize-fields.ts

```diff
@@ -201,5 +214,31 @@ export default function normalizeFields< Item >(
 			filterBy,
 			readOnly: field.readOnly ?? fieldTypeDefinition.readOnly ?? false,
 		};
+
+		if ( field.type === 'date' ) {
+			return {
+				...baseField,
+				type: 'date' as const,
+				format: {
+					date:
+						field.format?.date !== undefined &&
+						typeof field.format.date === 'string'
+							? field.format.date
```

Comment:
Should we have some kind of validation here and if the provided value is invalid, still apply the defaults? Same for `weekStartsOn`.

Reply:
Actually my comment is more valuable about `format`, where we could `try/catch` with `parse`? We can look into this in a follow up maybe.